rosette/rbl/rosette/namespace-util.rbl

```diff
@@ -0,0 +1,112 @@
+;;; TODO: Remove unused methods in Tuplespace and add tests for the rest
+
+;;; TODO: Add tag for whether persistent
+;;; TODO: Translate match in compiler
+
+(random-number-init)
+;;; The debugSymbol is inserted so that the output is easier to read
+(defOprn generateFresh)
+(defProc (generateFresh debugSymbol)
+  (->symbol
+   (string-concat "Rholang-" debugSymbol "-" (->string (random-number)))))
+
+(defOprn append)
+(defPure Tuple (append item)
+  (concat (self) [item]))
+
+(defOprn string-length<)
+(defOprn string-length>=)
+(defPure String (string-length< other)
+  (< (string-length (self)) (string-length other)))
+(defPure String (string-length>= other)
+  (not (string-length< (self) other)))
+
+(defOprn symbol-length<)
+(defOprn symbol-length>=)
+(defPure Symbol (symbol-length< other)
+  (string-length< (->string (self)) (->string other)))
+(defPure Symbol (symbol-length>= other)
+  (string-length>= (->string (self)) (->string other)))
+
+;;; Deletes element at position i in Tuple
+(defOprn delete-i)
+(defPure Tuple (delete-i p)
+  (concat (prim-sub-object (self) 0 p)
+          (prim-sub-object (self) (fx+ p 1)
+                           (fx- (prim-size (self)) (fx+ p 1)))))
+
+(defOprn replace-i)
+(defPure Tuple (replace-i p item)
+  (concat
+   (prim-sub-object (self) 0 p)
+   [item]
+   (prim-sub-object (self) (fx+ p 1)
+                    (fx- (prim-size (self)) (fx+ p 1)))))
+
+(defOprn symbol-list-concat)
+(defPure Tuple (symbol-list-concat) (->symbol (fold (self)
+                                                    (proc [e acc k]
+                                                      (k (string-concat acc (->string e)))) "")))
+
+;;; TODO: Figure out a more general flatten method
+(defOprn append-tuples)
+(defPure Tuple (append-tuples) (fold (self) (proc [e acc k] (k (tuple-concat acc e))) []))
+
+(defOprn twoZip)
+(defPure Tuple (twoZip)
```

Comment:
Yeah some of these are now outdated/unused. I was hoping to get away with the TODO at the top of this file for now and submit the cleanup as another PR, but I think it will save us words if I just cleaned this up now.

Reply:
To be clear, these helper functions show up even though I didn't touch them because I split the original space.rbl into two.

Reply:
I owe you and @kirkwood an apology for making you two read four helper functions (that aren't used) that took me 5 minutes just now to remove. I wish Github had a way of not showing only file move changes.